mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -371,6 +374,20 @@ class PreferencesActivity : AbstractBaseActivity() {
             }
         }
 
+        override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+            when (requestCode) {
+                PERMISSIONS_REQUEST_READ_PHONE_STATE -> {
+                    if (grantResults.isEmpty() || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                        context!!.showToast(R.string.settings_phone_state_permission_denied)
```

Comment:
Maybe rather `context?.showToast()`? In the unlikely case of the fragment being detached, it seems fine to not show a toast, better than crashing at least.